pkg/broker/broker.go

```diff
@@ -850,10 +859,194 @@ func (a AnsibleBroker) Unbind(
+	//type UpdateRequest struct {
+
+	//-> PreviousValues
+	//  -> OrganizationID    uuid.UUID
+	//  -> SpaceID           uuid.UUID
+	//   Used for determining where this service should be provisioned. Analogous to
+	//   OCP's namespaces and projects. Re: OrganizationID, spec mentions
+	//   "Most brokers will not use this field, it could be helpful in determining
+	//   the data placement or applying custom business rules"
+	//   -> PlanID            uuid.UUID
+	//   -> ServiceID         uuid.UUID
+	// ServiceID maps directly to a Spec.Id found in etcd. Can pull Spec via
+	// Dao::GetSpec(id string)
+
+	//-> Parameters        map[string]string
+	// User provided configuration answers for the AnsibleApp
+
+	// -> AcceptsIncomplete bool
+	// true indicates both the SC and the requesting client (sc client). If param
+	// is not included in the req, and the broker can only provision an instance of
+	// the request plan asyncronously, broker should reject with a 422
+	// NOTE: Spec.Async should indicate what level of async support is available for
+	// a given ansible app
+
+	//}
+
+	// Summary:
+	// For our purposes right now, the ServiceID and the Params should be enough to
+	// Update an ansible app.
+	////////////////////////////////////////////////////////////
+	// Update Flow
+	// -> Retrieve Spec from etcd (if missing, 400, this returns err missing)
+	// -> Retrieve Instance from etcd (if missing, 400, this returns err missing)
+	// -> TODO: Check to see if the spec supports or requires async, and reconcile
+	//    need a typed error condition so the REST server knows correct response
+	//    depending on the scenario
+	//    (async requested, unsupported, 422)
+	//    (async not requested, required, ?)
+	// -> Update entry in /instance, ID'd by instance. Value should be Instance type
+	//    Purpose is to make sure everything neeed to deprovision is available
+	//    in persistence.
+	// -> Update!
+	////////////////////////////////////////////////////////////
+
+	/*
+	   dao GET returns error strings like CODE: message (entity) [#]
+	   dao SetServiceInstance returns what error?
+	   dao.SetState returns what error?
+	   Provision returns what error?
+	   SetExtractedCredentials returns what error?
+
+	   broker
+	   * normal synchronous return UpdateResponse
+	   * normal async return UpdateResponse
+
+	   handler returns the following
+	   * synchronous update return 201 created
+	   * instance already exists with IDENTICAL parameters to existing instance, 200 OK
+	   * async provision 202 Accepted
+	   * if only support async and no accepts_incomplete=true passed in, 422 Unprocessable entity
+
+	*/
+
+	var err error
+
+	si, err := a.dao.GetServiceInstance(instanceUUID.String())
+	if err != nil {
+		a.log.Debug("Error retrieving instance")
+		return nil, ErrorNotFound
+	}
+
+	// If no plan was specified we'll continue with the current plan
+	if req.PlanID == "" {
+		req.PlanID = (*si.Parameters)["_apb_plan_id"].(string)
```

Comment:
One other note now that I look at it again: There's a const that should be used instead of the string literal `_apb_plan_id` for this key: https://github.com/openshift/ansible-service-broker/blob/master/pkg/broker/types.go#L35

So any `"_apb_plan_id"` should get updated to be `planParameterKey` instead.